.github/workflows/java-maven-openjdk-codeql.yml

```diff
@@ -0,0 +1,62 @@
+name: "CodeQL"
+
+on:
+  workflow_call:
+    inputs:
+      runs-on:
+        description: |
+          The type of machine to run the job on. Must be provided as a stringified list (e.g. `runs-on: '["ubuntu-latest","self-hosted"]'`)
+        required: true
+        type: string
+      mvn-version:
+        required: false
+        type: number
+        default: 3.9
+      mvn-arguments:
+        required: false
+        type: string
+        default: '--also-make --batch-mode --strict-checksums --update-snapshots -Dmaven.gitcommitid.skip=true'
+      jdk-version:
+        required: true
+        type: number
+
+jobs:
+  analyze-java:
+    name: Analyze Java
+
+    runs-on: ${{ fromJson(inputs.runs-on) }}
+
+    container:
+      image: maven:${{ inputs.mvn-version }}-amazoncorretto-${{ inputs.jdk-version }}-debian:latest
```

Comment:
What's the incentive to use this custom docker image instead of https://github.com/actions/setup-java ?

Reply:
Nothing, I based this workflow on the other one we have with jdk 11 hardcoded. setup-java could be more flexible but I'm not sure we need it? Also, it doesn't seem to support jdk 20.

Reply:
The setup-java action does not setup Maven, we would need to add a new action to our "trusted" actions list to setup Maven.

- https://github.com/marketplace/actions/setup-maven-action

Using actions allows our workflows to run on EKS runners instead of using EC2 runners. EKS runners are less expensive.

Reply:
So ideally we should do it JP so it can run on eks runners? Just to make sure I understand

Reply:
Keep it like this, less complexity 😅 Running on EKS brings some other bugs.